### PR TITLE
[server] Fix ingestion isolation heartbeat request timeout

### PR DESCRIPTION
**Pull request by sixpluszero**
## [server] Fix ingestion isolation heartbeat request timeout
This PR fixes a bug: ingestion isolation heartbeat timeout uses wrong time unit so it won't timeout until 10000 seconds rather than 10 seconds. If request stuck for some reason, the forked process will exit first due to correct timeout detection. However, main process won't detect until 10000s and it will make following ingestion all failed.

This PR also makes following adjustments:
1. Adjust periodic requests(Heartbeat/Metric) timeout to 5 seconds, so it won't be blocking forever. It is expected to complete fast, unless soemthing wrong happened and we should be able to detect it in the short period.
2. Do not update heartbeat timestamp in metric request. Only update it in heartbeat request so it is easier for debugging.
3. Adjust logging to make it more clear.
4. Align default heartbeat request timeout to 180s, as it has been adopted in EI config long time ago.

## How was this PR tested?
Internal CI
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ public IsolatedIngestionServer(String configPath) throws FileNotFoundException {
     this.configLoader = new VeniceConfigLoader(loadedVeniceProperties, loadedVeniceProperties, kafkaClusterMap);
     this.servicePort = configLoader.getVeniceServerConfig().getIngestionServicePort();
     this.heartbeatTimeoutMs = configLoader.getCombinedProperties()
-        .getLong(SERVER_INGESTION_ISOLATION_HEARTBEAT_TIMEOUT_MS, 60 * Time.MS_PER_SECOND);
+        .getLong(SERVER_INGESTION_ISOLATION_HEARTBEAT_TIMEOUT_MS, 180 * Time.MS_PER_SECOND);
```
[the remark this answers is not in the document]
Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java:157]: No, this is the heartbeat timeout for restarting fork process or for the fork process to quit itself. It means if no heartbeat has been updated for this long, both processes will act.

[the remark this answers is not in the document]
Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java:157]: I changed each HB/Metric request timeout to 5s so no indefinite waiting.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionMonitorService.java [hunk 2]
```diff
@@ -296,33 +296,38 @@ private void resumeOngoingIngestionTasks() {
...
-          currentTimeMillis,
+          "Heartbeat request to forked process issued at {}, failed at {}, latest successful timestamp: {}",
+          responseTimestamp,
+          requestTimestamp,
           latestHeartbeatTimestamp);
     }
 
     if (latestHeartbeatTimestamp != -1) {
-      if ((currentTimeMillis - latestHeartbeatTimestamp) > heartbeatTimeoutMs) {
+      if ((requestTimestamp - latestHeartbeatTimestamp) > heartbeatTimeoutMs) {
```
[the remark this answers is not in the document]
Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionMonitorService.java:330]: This is more than 3 times. HB is 10s per time, timeout is 180s. We actually allow about 18times failure.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaojieliu) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java:157]: I see, so this config is not for one heart-beat request.
Can you add some javadoc to highlight this is not for a single heart-beat request?

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionMonitorService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (gaojieliu) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionMonitorService.java:330]: Got it.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java:161]: Sure, just added java doc for both main and forked process.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionRequestClient.java [hunk 3]
```diff
@@ -40,7 +39,7 @@
 public class MainIngestionRequestClient implements Closeable {
   private static final Logger LOGGER = LogManager.getLogger(MainIngestionRequestClient.class);
   private static final int REQUEST_MAX_ATTEMPT = 10;
-  private static final int HEARTBEAT_REQUEST_TIMEOUT_MS = 10 * Time.MS_PER_SECOND;
+  private static final int PERIODIC_REQUEST_TIMEOUT_SECONDS = 5;
```
Comment (gaojieliu) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionRequestClient.java:42]: Nit:
Can we rename it to be `PERIODIC_HEART_BEAT_REQUEST_TIMEOUT_SECONDS` to be more specific?

Comment (gaojieliu) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionRequestClient.java:42]: Got it.

**Review by gaojieliu — APPROVED** [review]

**Pull request merged**